test/pylib/scylla_cluster.py

```diff
@@ -698,6 +721,33 @@ def update_config(self, server_id: ServerNum, key: str, value: object) -> Action
         self.servers[server_id].update_config(key, value)
         return ScyllaCluster.ActionReturn(success=True)
 
+    async def change_ip(self, server_id: ServerNum) -> ActionReturn:
+        """Lease a new IP address and update conf/scylla.yaml with it. The
+        original IP is released at the end of the test to avoid recycle
+        within the existing cluster (scylla currently doesn't support
+        it)."""
+        if server_id not in self.servers:
+            return ScyllaCluster.ActionReturn(success=False, msg=f"Server {server_id} unknown")
+        server = self.servers[server_id]
+        if server.is_running:
+            msg = f"Server {server_id} is running: stop fist it and then change its ip"
+            return ScyllaCluster.ActionReturn(success=False, msg=msg)
+        self.is_dirty = True
+        ip_addr = IPAddress(await self.host_registry.lease_host())
+        self.leased_ips.add(ip_addr)
+        logging.info("Cluster %s changed server %s IP from %s to %s", self.name,
+                     server_id, server.ip_addr, ip_addr)
+        old_ip_addr = server.ip_addr
+        server.change_ip(ip_addr)
```

Comment:
`ScyllaServer.change_ip()` can raise an exception, it would be useful for debugging to catch it and `return ScyllaCluster.ActionReturn(success=False, msg=...)`

Reply:
The only thing that can raise is  write_config_file. I checked other calls, and they don't intercept exceptions, such as update_config(), server_start(), server_stop(). I think we need to be systematic about exceptions, and have a general mechanism for all of them, rather than handle each one with try/catch. We could use something likehttps://github.com/yuvalherziger/aiohttp-catcher for the purpose.